### PR TITLE
Add info logs for delete operations

### PR DESCRIPTION
**Pull request by shekhars-li**
- Update some debug/trace logs to info logs to better track delete operations. 
- Check to verify that a file is not added in files deletes as well as files added/deleted of `DirIndex`

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java [hunk 1]
```diff
@@ -173,4 +182,5 @@ public String toString() {
           '}';
     }
   }
+
```
Comment (dxichen) [inline samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java:185]: Remove extra line

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java [hunk 2]
```diff
@@ -48,6 +51,12 @@ public DirIndex(String dirName,
     Preconditions.checkNotNull(filesRemoved);
     Preconditions.checkNotNull(subDirsPresent);
     Preconditions.checkNotNull(subDirsRemoved);
+    // check to validate that a file is not present in file removed and file present list
```
Comment (dxichen) [inline samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java:54]: When could this happen? What scenario are we defending against?

Nit: s/check/Check

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java (hunk 1 above)
Comment (shekhars-li) [reply samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java]: Fixed. Thanks.

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java (hunk 2 above)
Comment (shekhars-li) [reply samza-core/src/main/java/org/apache/samza/storage/blobstore/index/DirIndex.java]: This is basic sanity check, similar to what we are doing in [DirDiff](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/storage/blobstore/diff/DirDiff.java#L84).

**Comment by prateekm** [comment]
@shekhars-li Do you also want to add/increase log level in TaskInstance where cleanup is called (during init / post-commit)?Also, maybe log the task checkpoint on commit?

**Review by prateekm — APPROVED** [review]

**Pull request merged**